src/main/java/co/cask/gcp/bigquery/BigQuerySink.java

```diff
@@ -519,21 +520,22 @@ private Table getBigqueryTable(BigQuerySinkConfig config) throws IOException {
   }
 
   private void matchSchema(BigQuerySinkConfig config, FieldList bqFields, List<Schema.Field> outputSchemaFields) {
-    Map<Schema.Type, LegacySQLTypeName> typeMap = new HashMap<>();
-    typeMap.put(Schema.Type.INT, LegacySQLTypeName.INTEGER);
-    typeMap.put(Schema.Type.LONG, LegacySQLTypeName.INTEGER);
-    typeMap.put(Schema.Type.STRING, LegacySQLTypeName.STRING);
-    typeMap.put(Schema.Type.FLOAT, LegacySQLTypeName.FLOAT);
-    typeMap.put(Schema.Type.DOUBLE, LegacySQLTypeName.FLOAT);
-    typeMap.put(Schema.Type.BOOLEAN, LegacySQLTypeName.BOOLEAN);
-    typeMap.put(Schema.Type.BYTES, LegacySQLTypeName.BYTES);
-
-    Map<Schema.LogicalType, LegacySQLTypeName> logicalTypeMap = new HashMap<>();
-    logicalTypeMap.put(Schema.LogicalType.DATE, LegacySQLTypeName.DATE);
-    logicalTypeMap.put(Schema.LogicalType.TIME_MILLIS, LegacySQLTypeName.TIME);
-    logicalTypeMap.put(Schema.LogicalType.TIME_MICROS, LegacySQLTypeName.TIME);
-    logicalTypeMap.put(Schema.LogicalType.TIMESTAMP_MILLIS, LegacySQLTypeName.TIMESTAMP);
-    logicalTypeMap.put(Schema.LogicalType.TIMESTAMP_MICROS, LegacySQLTypeName.TIMESTAMP);
+    Map<Schema.Type, LegacySQLTypeName> typeMap = ImmutableMap.<Schema.Type, LegacySQLTypeName>builder()
```

Comment:
By constant I meant these can be private static final in the class so that we don't need to build them when the method is called.